src/services/event.js

```diff
@@ -5,98 +5,124 @@
+/**
+ * Gets all events from the backend, and then formats and sorts them
+ * @returns {Event[]} All events
+ */
+const getAllEvents = async () => {
+  const allEvents = await http.get('events/25Live/All');
+  return allEvents.map((e) => formatevent(e)).sort(sortEventsByTime);
+};
 
-const getAllCLAWEvents = () => http.get('/events/25Live/CLAW');
+/**
+ * Gets upcoming CL&W events and formats them for display
+ * TODO: Unused. Consider removing
+ * @returns {Event[]} upcoming CL&W events
+ */
+const getCLWEvents = async () => {
+  const allEvents = await http.get('events/25Live/CLAW');
+  const now = Date.now();
+  return allEvents
+    .filter((e) => new Date(e.Occurrences[0].StartDate).getTime() > now)
+    .map((e) => formatevent(e))
+    .sort(sortEventsByTime);
+};
 
-const getAllGuestEvents = () => http.get('events/25Live/Public');
+/**
+ * Gets all public events from the backend, and then formats and sorts them
+ * @returns {Event[]} All events
+ */
+const getAllGuestEvents = async () => {
+  const allGuest = await http.get('events/25Live/Public');
+  return allGuest.map((e) => formatevent(e)).sort(sortEventsByTime);
+};
+
+/**
+ * Get chapel events attended by the user during the current term
+ * @return {AttendedEvent[]} all CL&W events attended by the user, formatted and sorted
+ */
+const getAttendedChapelEvents = async () => {
+  const termCode = session.getTermCode();
+  let attendedEvents = await http.get(`events/chapel/${termCode}`);
+  return attendedEvents.map((e) => formatevent(e)).sort(sortAtndEventsByTime);
+};
 
+/**
+ *  Format an event for display on the front end
+ * @param {Event} event The event to format
+ * @returns {Event} The formatted event
+ */
 function formatevent(event) {
-  let beginTime;
-  let endTime;
-  if (event.Occurrences[0] && event.Occurrences[0][0] && event.Occurrences[0][1]) {
-    beginTime = DateTime.fromISO(event.Occurrences[0][0]).toFormat('t');
-    endTime = DateTime.fromISO(event.Occurrences[0][1]).toFormat('t');
+  let formattedEvent = { ...event };
+  if (event.Occurrences[0]) {
+    let beginTime = DateTime.fromISO(event.Occurrences[0].StartDate).toFormat('t');
+    let endTime = DateTime.fromISO(event.Occurrences[0].EndDate).toFormat('t');
+    formattedEvent.timeRange = `${beginTime} - ${endTime}`;
+    formattedEvent.date = DateTime.fromISO(event.Occurrences[0].StartDate).toFormat('LLL d, yyyy');
   }
-  const timeRange = `${beginTime} - ${endTime}`;
-  event.timeRange = timeRange;
 
-  let date;
-  if (event.Occurrences[0] && event.Occurrences[0][0]) {
-    date = DateTime.fromISO(event.Occurrences[0][0]).toFormat('LLL d, yyyy');
-  }
-  event.date = date;
+  formattedEvent.title = event.Event_Title || event.Event_Name;
 
-  let title;
-  if (event.Event_Title === '') {
-    title = event.Event_Name;
-  } else {
-    title = event.Event_Title;
-  }
-  event.title = title;
+  formattedEvent.location = event.Occurrences[0].Location || 'No Location Listed';
```

Comment:
Good catch! This would've thrown an error for any events with no Occurrences property or with an empty Occurrences array. I'm not sure there are any events with 0 Occurrences, but much better to build it robustly just in case. I've fixed this issue.